File: Python/001. Two Sum.py
"""
Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

Example:
    Given nums = [2, 7, 11, 15], target = 9,

    Because nums[0] + nums[1] = 2 + 7 = 9,
    return [0, 1].
"""


class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if (nums[i] + nums[j] == target):
                    result = (i, j)

        return list(result)


if __name__ == "__main__":
    solution = Solution()
    print(solution.twoSum([3, 2, 4], 6))
    print(solution.twoSum([3, 3], 6))

File: C++/001. Two Sum.cpp
//
// Created by 王阳 on 2018/3/11.
//

/**
    Given an array of integers, return indices of the two numbers such that they add up
    to a specific target. You may assume that each input would have exactly one solution,
    and you may not use the same element twice.

    Example:
        Given nums = [2, 7, 11, 15], target = 9,

    Because nums[0] + nums[1] = 2 + 7 = 9,
    return [0, 1].
 */
#include "header.h"

class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        vector<int> res;
        for (int i = 0; i < nums.size(); i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] + nums[j] == target) {
                    res.push_back(i);
                    res.push_back(j);
                    return res;
                }
            }
        }
        return res;
    }
};

int main() {
    return 0;
}